velox/functions/lib/RegistrationHelpers.h

```diff
@@ -83,6 +83,36 @@ void registerUnaryNumeric(const std::vector<std::string>& aliases) {
   registerUnaryFloatingPoint<T>(aliases);
 }
 
+#define REGISTER_NUMERIC_FUNCTIONS(functionName, prefix) \
+  {                                                      \
+    functionName<int8_t>(prefix);                        \
+    functionName<int16_t>(prefix);                       \
+    functionName<int32_t>(prefix);                       \
+    functionName<int64_t>(prefix);                       \
+    functionName<int128_t>(prefix);                      \
+    functionName<float>(prefix);                         \
+    functionName<double>(prefix);                        \
+  }
+
+#define REGISTER_SCALAR_FUNCTIONS(functionName, prefix) \
+  {                                                     \
+    REGISTER_NUMERIC_FUNCTIONS(functionName, prefix)    \
+    functionName<Varchar>(prefix);                      \
+    functionName<Varbinary>(prefix);                    \
+    functionName<bool>(prefix);                         \
+    functionName<Timestamp>(prefix);                    \
+    functionName<Date>(prefix);                         \
```

Comment:
Could call the `REGISTER_SCALAR_FUNCTIONS_WITHOUT_VARCHAR` and add a varchar registration.